### PR TITLE
Rewrite of the SCM optimization tip box

### LINKED ISSUE #1116: Only 'create' command supported by SCM copy optimization
We should update docs here (the `tip` box): https://docs.conan.io/en/1.6/creating_packages/package_repo.html#capturing-the-remote-and-commit-from-git-scm-experimental

Pointed out in this comment 👍  https://github.com/conan-io/conan/issues/4732#issue-420484150

### PR DESCRIPTION
**Pull request by danimtb**
closes #1116

### CODE REVIEW

File: creating_packages/package_repo.rst [hunk 1]
```diff
@@ -172,8 +172,8 @@ be valid too:
 
 .. tip::
 
-    While you are in the same computer (the same Conan cache), even when you have exported the recipe and
-    Conan has captured the absolute url and commit, Conan will store the local folder where your source code lives.
-    If you build your package locally, it will use the local repository (in the local folder) instead of the remote URL,
-    even if the local directory contains uncommitted changes.
-    This allows you to speed up the development of your packages when cloning from a local repository.
+    When doing a :command:`conan create` of a recipe using ``scm``, Conan will store a local path to the local source repository. Every time
+    you want to create a new package, the sources will not be downloaded from the remote repository but from the local one.
```
Comment (jgsogo) [inline creating_packages/package_repo.rst]: something like _"...every time you invoke the `create` command for this package..."_?

File: creating_packages/package_repo.rst [hunk 2]
```diff
@@ -172,8 +172,8 @@ be valid too:
 
 .. tip::
 
-    While you are in the same computer (the same Conan cache), even when you have exported the recipe and
-    Conan has captured the absolute url and commit, Conan will store the local folder where your source code lives.
-    If you build your package locally, it will use the local repository (in the local folder) instead of the remote URL,
-    even if the local directory contains uncommitted changes.
-    This allows you to speed up the development of your packages when cloning from a local repository.
+    When doing a :command:`conan create` of a recipe using ``scm``, Conan will store a local path to the local source repository. Every time
```
Comment (jgsogo) [inline creating_packages/package_repo.rst]: Maybe rephrase a little bit: _"Conan will store in a file the path to the local source repository"_.

File: creating_packages/package_repo.rst [hunk 3]
```diff
@@ -172,8 +172,8 @@ be valid too:
...
+    you want to create a new package, the sources will not be downloaded from the remote repository but from the local one.
+
+    This allows to build packages making changes to the source code without the need of committing them and pushing them to the remote
+    repository. This convenient to speed up the development of your packages when cloning from a local repository.
```
Comment (jgsogo) [inline creating_packages/package_repo.rst]: A warning here about the _not-reproducibility_ of packages generated using this _trick_ if there are changes not committed in thee repo?

Comment (danimtb) [reply creating_packages/package_repo.rst]: A warning inside a tip? 🤣 Let's see what I can do

Comment (jgsogo) [reply creating_packages/package_repo.rst]: A mean, warning the user about it, not a _warning box_ 🙈

File: creating_packages/package_repo.rst [hunk 4]
```diff
@@ -172,8 +172,11 @@ be valid too:
 
 .. tip::
 
-    While you are in the same computer (the same Conan cache), even when you have exported the recipe and
-    Conan has captured the absolute url and commit, Conan will store the local folder where your source code lives.
-    If you build your package locally, it will use the local repository (in the local folder) instead of the remote URL,
-    even if the local directory contains uncommitted changes.
-    This allows you to speed up the development of your packages when cloning from a local repository.
+    When doing a :command:`conan create` of a recipe using ``scm``, Conan will save the path to the local source repository. Every time the
+    :command:`conan create` command is ivoked, the sources will not be downloaded from the remote repository but from the local one.
```
Comment (jgsogo) [inline creating_packages/package_repo.rst]:
```suggestion
    :command:`conan create` command is invoked, the sources will not be downloaded from the remote repository but copied from the local directory.
```

Comment (danimtb) [reply creating_packages/package_repo.rst]: ouch
